src/NuGet.Core/NuGet.Protocol/Plugins/PluginPackageReader.cs

```diff
@@ -1032,7 +1032,7 @@ protected override void Dispose(bool disposing)
             var response = await _plugin.Connection.SendRequestAndReceiveResponseAsync<CopyFilesInPackageRequest, CopyFilesInPackageResponse>(
                 MessageMethod.CopyFilesInPackage,
                 payload,
-                CancellationToken.None);
+                cancellationToken);
```

Comment:
Then, the caller has to figure out what to do with cancelled operation and the 'corrupted' state.

First commit in history of this file shows no usage of cancellationToken. So, I thought it might be useful to pass cancelation token

https://github.com/NuGet/NuGet.Client/blob/870340673ba75a95407ab55ce1f46f57fe9bf25b/src/NuGet.Core/NuGet.Protocol/Plugins/PluginPackageReader.cs#L1037

Do you think current test GetStreamAsync_ThrowsIfCancelled does not cover this scenario?

Reply:
PR available to restore original method: https://github.com/NuGet/NuGet.Client/pull/4444